data/mods/gen1/random-teams.ts

```diff
@@ -301,9 +269,12 @@ export class RandomGen1Teams extends RandomGen2Teams {
 			moves.add(this.sample(data.exclusiveMoves));
 		}
 
-		// Add the mandatory move. SD Mew and Amnesia Snorlax are exceptions.
-		if (moves.size < this.maxMoveCount && data.essentialMove) {
-			moves.add(data.essentialMove);
+		// Add the mandatory moves.
```

Comment:
I would prefer a comment indicating why "mandatory moves" are generated after "semi-mandatory moves", or to swap them around if there's no real reason why. It's not caused by this PR however, so not a big issue right now.

Reply:
I added a comment explaining this. 

Semi-mandatory moves are chosen before mandatory moves so that combos with three moves can choose an exclusive move for the fourth move. This slightly increases setdev flexibility without any loss in functionality. (Choosing a mandatory move after a three move combo would not offer additional possibilities, as we could easily include the mandatory move in the combo).

Previously, this was used for Mew's Swords Dance combo which contained three moves, so that the last move could be a roll between Softboiled and Explosion. There are no longer any three move combos, but I'd like to preserve the possibility for the future.